File: src/main/java/goodroad/model/RouteRequest.java
package goodroad.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import java.util.List;
import java.util.ArrayList;

@Data
public class RouteRequest {
    private String start;
    private String end;

    @JsonProperty("user_id")
    private String userId;

    @JsonProperty("max_stairs")
    private Integer maxStairsCount;

    @JsonProperty("max_slope")
    private Double maxSlopeAngle;

    @JsonProperty("max_curb_height")
    private Integer maxCurbHeight;

    @JsonProperty("min_path_width")
    private Integer minPathWidth;

    @JsonProperty("avoid_stairs")
    private boolean avoidStairs = false;

    @JsonProperty("need_ramp")
    private boolean needRamp = false;

    @JsonProperty("avoid_bad_road")
    private boolean avoidBadRoad = false;

    @JsonProperty("avoid_surfaces")
    private List<String> avoidSurfaceTypes = new ArrayList<>();

    @JsonProperty("obstacle_policies")
    private List<RouteObstaclePolicy> obstaclePolicies = new ArrayList<>();

    private String locale = "ru";

    @JsonProperty("alternatives")
    private boolean needAlternatives = true;

    @JsonProperty("points_encoded")
    private boolean pointsEncoded = true;

    @Data
    public static class RouteObstaclePolicy {
        @JsonProperty("obstacle_type")
        private String obstacleType;

        @JsonProperty("max_allowed_severity")
        private Short maxAllowedSeverity;
    }
}

File: src/main/kotlin/com/GoodRoad/model/RouteRequest.kt
package com.GoodRoad.model

import com.fasterxml.jackson.annotation.JsonProperty

data class RouteRequest(
    val start: String,              // (lat,lon)
    val end: String,                // (lat,lon)
    @JsonProperty("user_id")
    val userId: String,

    @JsonProperty("max_stairs")
    val maxStairsCount: Int? = null,        // сколько ступенек максимум
    @JsonProperty("max_slope")
    val maxSlopeAngle: Double? = null,       // макс угол уклона
    @JsonProperty("max_curb_height")
    val maxCurbHeight: Int? = null,          // макс высота бордюра
    @JsonProperty("min_path_width")
    val minPathWidth: Int? = null,            // мин ширина прохода

    @JsonProperty("avoid_stairs")
    val avoidStairs: Boolean = false,            // избегать лестниц
    @JsonProperty("need_ramp")
    val needRamp: Boolean = false,                // нужен пандус
    @JsonProperty("avoid_bad_road")
    val avoidBadRoad: Boolean = false,            // избегать плохих дорог

    // какие поверхности избегать
    @JsonProperty("avoid_surfaces")
    val avoidSurfaceTypes: List<String> = emptyList(),  // "SAND", "GRAVEL"

    val locale: String = "ru",                    // язык инструкций
    @JsonProperty("alternatives")
    val needAlternatives: Boolean = true,          // нужны ли альтернативы
    @JsonProperty("points_encoded")
    val pointsEncoded: Boolean = true
)